### PR TITLE
Added TypeScript type declarations (index.d.ts)

### PR DESCRIPTION
**Pull request by abrasher**
I wanted to use typescript with your library so I created a type declaration file for this plugin.

Type declarations are sourced from your code and the documentation.

I believe updateDefaults take in the same options as Vue.use but not sure about that.

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+  /**
+   *  Position of the toast on the screen.
+   * 
+   *  Can be any of top-right, top-center, top-left, bottom-right, bottom-center, bottom-left.
+   */ 
+  draggable?: boolean;
+  /**
+   *  By how much of the toast width in percent (0 to 1) it must be dragged before being dismissed.
+   */
+  draggablePercent?: boolean;
```
Comment (Maronato) [inline index.d.ts]: `draggablePercent` is actually a float that goes from 0 to 1, so `number` is probably the right type
```suggestion
  draggablePercent?: number;
```

File: index.d.ts [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+  }
+  
+  interface VueConstructor {
+    $toast: ToastModule;
+  }
+}
+
+export default class Toast {
+  static install: PluginFunction<never>;
+}
```
Comment (Maronato) [inline index.d.ts]: Missing a newline over here

**Comment by Maronato** [comment]
Amazing! I was actually writing the type declarations right now to address #4, but you beat me to it and did a far superior job. Your assumption about `updateDefaults` is correct.

**Comment by abrasher** [comment]
Oops yeah I missed that

**Comment by abrasher** [comment]
Oops yeah I missed that

**Comment by Maronato** [comment]
@abrasher no need to close the PR! Are you going to open another one?

**Comment by abrasher** [comment]
Yeah I don't know what I

**Comment by Maronato** [comment]
Excellent :)

**Comment by iliyaZelenko** [comment]
@abrasher good job, this will help me. 👍

**Pull request merged**